resources/splunk_app.rb

```diff
@@ -136,3 +148,41 @@ def config_scope
     apply_config
   end
 end
+
+class PackagedApp < SplunkApp
+  property :source_url, String, required: true
+
+  resource_name :splunk_app_package
+
+  def package_path
+    return @package_path if @package_path
+    file_cache = Pathname.new(Chef::Config['file_cache_path'])
+    @package_path = file_cache.join(CernerSplunk::PathHelpers.filename_from_url(source_url).gsub(/.spl$/, '.tgz'))
+  end
+
+  action :install do
+    return unless !version || changed?(:version)
+
+    remote_file package_path.to_s do
```

Comment:
> How would we unexpectedly get a prerelease version given a specific source url?

Easy... upload something to the wrong spot, or have an out of date working copy and build/upload. There wasn't and still really isn't much in the way of an ecosystem for building / packaging apps sadly, so I wasn't assuming any protections from an app build system in this

> Properly handling that with the future upgrade strategy would complicate things though. It would be better to fail on that case.

This is the core of why we have the fail for local content. I would support a flag saying that "no we expect there to be local content in this package" that is default set to fail instead of pass but yeah.